src/dvc_objects/fs/implementations/gdrive.py

```diff
@@ -51,172 +51,75 @@ def __init__(self, **config):
         if not opts["host"]:
             raise ConfigError("Empty GDrive URL '{}'.".format(config["url"]))
 
-        self._bucket = opts["host"]
-        self._path = opts["path"].lstrip("/")
-        self._trash_only = config.get("gdrive_trash_only")
-        self._use_service_account = config.get("gdrive_use_service_account")
-        self._service_account_user_email = config.get(
-            "gdrive_service_account_user_email"
+        self._path = opts["host"] + opts["path"]
+
+        client_id = config.get(
+            "gdrive_client_id", self.DEFAULT_GDRIVE_CLIENT_ID
         )
-        self._service_account_json_file_path = config.get(
-            "gdrive_service_account_json_file_path"
+        client_secret = config.get(
+            "gdrive_client_secret", self.DEFAULT_GDRIVE_CLIENT_SECRET
         )
-        self._client_id = config.get("gdrive_client_id")
-        self._client_secret = config.get("gdrive_client_secret")
-        self._validate_config()
-
-        self._creds_path = config.get("gdrive_user_credentials_file")
-        tmp_dir = config.get("gdrive_credentials_tmp_dir")
-        if not self._creds_path and tmp_dir:
-            self._creds_path = os.path.join(
-                tmp_dir, self.DEFAULT_USER_CREDENTIALS_FILE
-            )
-
-    @classmethod
-    def _strip_protocol(cls, path):
-        from fsspec.utils import infer_storage_options
-
-        opts = infer_storage_options(path)
-
-        return "{host}{path}".format(**opts)
+        use_service_account = config.get("gdrive_use_service_account")
+        client_user_email = config.get("gdrive_service_account_user_email")
+        profile = config.get("profile")
 
-    def unstrip_protocol(self, path):
-        return f"gdrive://{path}"
+        if use_service_account:
+            client_json_file_path = config.get(
+                "gdrive_service_account_json_file_path"
+            )
+        else:
+            client_json_file_path = config.get("gdrive_user_credentials_file")
```

Comment:
To confirm: it still works, right? I'm trying to remember the logic - was it a relative path to the repo root? to the `.dvc/tmp`? or an absolute path ... I remember there was some special handling for this ... again, just trying to confirm if we need to change it in docs?

Reply:
At this point in the stack it is already resolved as an absolute path. There is a resolver in https://github.com/iterative/dvc/blob/main/dvc/config.py#L260

Reply:
sounds good, wanted to double check, thanks @efiop